app/src/main/kotlin/com/wire/android/ui/authentication/login/LoginViewModel.kt

```diff
@@ -35,31 +34,37 @@ open class LoginViewModel @Inject constructor(
     private val clientScopeProviderFactory: ClientScopeProvider.Factory,
     authServerConfigProvider: AuthServerConfigProvider
 ) : ViewModel() {
-    var loginState by mutableStateOf(
+    private val _loginStateFlow = MutableStateFlow(
         LoginState(
-            ssoCode = TextFieldValue(savedStateHandle.get(SSO_CODE_SAVED_STATE_KEY) ?: String.EMPTY),
+            ssoCode = TextFieldValue(savedStateHandle[SSO_CODE_SAVED_STATE_KEY] ?: String.EMPTY),
             userIdentifier = TextFieldValue(savedStateHandle[USER_IDENTIFIER_SAVED_STATE_KEY] ?: String.EMPTY),
             password = TextFieldValue(String.EMPTY)
         )
     )
-        protected set
+    val loginStateFlow: StateFlow<LoginState> = _loginStateFlow
+
+    protected fun updateLoginState(newState: LoginState) {
+        viewModelScope.launch { _loginStateFlow.emit(newState) }
+    }
 
     val serverConfig = authServerConfigProvider.authServer.value
 
     open fun updateSSOLoginError(error: LoginError) {
-        loginState = if (error is LoginError.None) {
-            loginState.copy(loginError = error)
+        val newState = if (error is LoginError.None) {
+            loginStateFlow.value.copy(loginError = error)
         } else {
-            loginState.copy(ssoLoginLoading = false, loginError = error).updateSSOLoginEnabled()
+            loginStateFlow.value.copy(ssoLoginLoading = false, loginError = error).updateSSOLoginEnabled()
         }
+        updateLoginState(newState)
     }
 
     open fun updateEmailLoginError(error: LoginError) {
-        loginState = if (error is LoginError.None) {
-            loginState.copy(loginError = error)
+        val newState = if (error is LoginError.None) {
+            loginStateFlow.value.copy(loginError = error)
         } else {
-            loginState.copy(emailLoginLoading = false, loginError = error).updateEmailLoginEnabled()
+            loginStateFlow.value.copy(emailLoginLoading = false, loginError = error).updateEmailLoginEnabled()
         }
+        updateLoginState(newState)
     }
```

Comment:
do not get me wrong, but I think the personal preferences to either use MUtableStateFlow of MutatbleStateOf, should not be taken as our collective decision, I have personally no strong on opinion which we are going to use, but this introduces chaos to a code, some places use mutableStateFlow some use MutableState, which one should we use ? I think if MutableStateFlow has some benefits over mutableStateOf we should take a decision on this during the collective and refactor piece by piece places we use the other option. Also ViewModel is not giong to be reused on iOS

Reply:
true story, it's better to discuss in Collective! 
I reverted `mutableStateFlow` changes for now :)